application/models/QuestionAttribute.php

```diff
@@ -219,16 +219,21 @@ public function setMultiple($iSid, $aQids, $aAttributesToUpdate, $aValidQuestion
      * --prepare an easier/smaller array to return
      *
      * @access public
-     * @param int $iQuestionID
+     * @param int|Question $q
      * @param string $sLanguage restrict to this language (if null $oQuestion->survey->allLanguages will be used)
      * @return array|false
      *
      * @throws CException throws exception if questiontype is null
      */
-    public function getQuestionAttributes($iQuestionID, $sLanguage = null)
+    public function getQuestionAttributes($q, $sLanguage = null)
```

Comment:
It wasn't possible to change all call locations to this method to always only pass object as input?

Reply:
We are currently working on performance issues. We could refactor the further references, but that would require a separate ticket and an approval

![image](https://github.com/LimeSurvey/LimeSurvey/assets/2805488/597eebcd-1013-4a43-a867-e5e326b877a4)

We can only change the parameter to accept question objects if we change and test all references. That would take a while and in the meantime the users would not get the benefits of our results. If we can create a ticket to refactor all references and then the parameter of this method, that would be great, I agree with you. But even then we should break it up to smaller PRs with the last one being the change of the parameter to not allow integers.

Reply:
Wow that's a lot of call sites.. Got it. :sweat_smile: